internal/gatewayapi/route.go

```diff
@@ -1176,6 +1178,17 @@ func (t *Translator) processDestination(backendRefContext BackendRefContext,
 			Endpoints:   endpoints,
 			AddressType: addrType,
 		}
+		// TODO: support mixed endpointslice address type for the same backendRef
```

Comment:
can we move this out of `switch` to make it common amongst all backend types ?

Reply:
The conditions is slightly different for `Backend` vs. the native K8s resources (depending on EndpointRouting toggle) and so is the message. I'll split the validation and status update.